OneCommunityGlobal/HighestGoodNetworkApp
# Leon and Oleksandr add interactive map

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108314277/565d2d3f-62d5-4c2b-99b2-673cd6c778b4)

## Related PRS 
This frontend PR is related to the #[599 ](https://github.com/OneCommunityGlobal/HGNRest/pull/599) backend PR.

## Main changes explained:
1. Added interactive map
2. Added new route
3. adding new users(Prior HGN data) functionality
4. editing/removing manually added users functionality
5. changed format of storing and saving location data of HGN users
6. editing HGN users info through map

## How to test:
1. set up backend
2. check out into current branch
3. do `npm install` and `npm run start:local` to run this PR locally
4. log in as owner user
5. go to reports -> team locations
6. add, edit, remove users with marker, list, and search
7. 
a) create new HGN user: and punch at least 10 tangible hours to the user     
b)  use existing one: go to user profile and update its location by getting timezone and saving 
8. find user on the map and try edit through marker/list/search 
9. go to user profile and compare
10. edit some info in the profile and compare to the map
11. login to any other user and confirm there is no way to add/edit/remove/search users

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108314277/6e90cd5f-1b10-4b89-ac45-880db331f864
